src/fastapi_app/dependencies.py

```diff
@@ -0,0 +1,138 @@
+        openai_embed_dimensions = int(os.getenv("AZURE_OPENAI_EMBED_DIMENSIONS", 1536))
+    else:
+        openai_embed_deployment = "text-embedding-ada-002"
+        openai_embed_model = os.getenv("OPENAICOM_EMBED_MODEL", "text-embedding-ada-002")
+        openai_embed_dimensions = int(os.getenv("OPENAICOM_EMBED_DIMENSIONS", 1536))
+    if OPENAI_CHAT_HOST == "azure":
+        openai_chat_deployment = os.getenv("AZURE_OPENAI_CHAT_DEPLOYMENT", "gpt-35-turbo")
+        openai_chat_model = os.getenv("AZURE_OPENAI_CHAT_MODEL", "gpt-35-turbo")
+    elif OPENAI_CHAT_HOST == "ollama":
+        openai_chat_deployment = "phi3:3.8b"
+        openai_chat_model = os.getenv("OLLAMA_CHAT_MODEL", "phi3:3.8b")
+    else:
+        openai_chat_deployment = "gpt-3.5-turbo"
+        openai_chat_model = os.getenv("OPENAICOM_CHAT_MODEL", "gpt-3.5-turbo")
+    return FastAPIAppContext(
+        openai_chat_model=openai_chat_model,
+        openai_embed_model=openai_embed_model,
+        openai_embed_dimensions=openai_embed_dimensions,
+        openai_chat_deployment=openai_chat_deployment,
+        openai_embed_deployment=openai_embed_deployment,
+    )
+
+
+async def get_azure_credentials() -> azure.identity.DefaultAzureCredential | azure.identity.ManagedIdentityCredential:
+    azure_credential: azure.identity.DefaultAzureCredential | azure.identity.ManagedIdentityCredential
+    try:
+        if client_id := os.getenv("APP_IDENTITY_ID"):
+            # Authenticate using a user-assigned managed identity on Azure
+            # See web.bicep for value of APP_IDENTITY_ID
+            logger.info(
+                "Using managed identity for client ID %s",
+                client_id,
+            )
+            azure_credential = azure.identity.ManagedIdentityCredential(client_id=client_id)
+        else:
+            azure_credential = azure.identity.DefaultAzureCredential()
+        return azure_credential
+    except Exception as e:
+        logger.warning("Failed to authenticate to Azure: %s", e)
+        raise e
+
+
+async def create_async_sessionmaker(engine: AsyncEngine) -> async_sessionmaker[AsyncSession]:
+    """Get the agent database"""
+    return async_sessionmaker(
+        engine,
+        expire_on_commit=False,
+        autoflush=False,
+    )
+
+
+async def get_async_sessionmaker(
+    request: Request,
+) -> AsyncGenerator[async_sessionmaker[AsyncSession], None]:
+    yield request.state.sessionmaker
+
+
+async def get_context(
+    request: Request,
+) -> FastAPIAppContext:
+    return request.state.context
+
+
+async def get_async_db_session(
+    sessionmaker: Annotated[async_sessionmaker[AsyncSession], Depends(get_async_sessionmaker)],
+) -> AsyncGenerator[AsyncSession, None]:
+    async with sessionmaker() as session:
+        try:
+            yield session
+        except:
+            await session.rollback()
+            raise
+        else:
+            await session.commit()
```

Comment:
Just curious, where did you see this particular try/except/else pattern? Was this recommended in the SQLAlchemy docs? We're using Depends with SQLModel sessions in another repo as well, but we only have "yield session" there, wondering if we should change it.